src/Clients/BaseApiClient.php

```diff
@@ -32,6 +36,15 @@ public function __construct(private $apiKey, private $apiSecret)
         $this->accessToken = $this->getAccessToken();
     }
 
+    public function retryPolicy(callable $retryPolicy, int $retryTimes = 1, int $retryInterval = 300): self
```

Comment:
O que acha de `withRetry()`, pra ficar tipo:

```php
$response = $client->withRetry(fn() => true)->get('/api')->json();
```

```suggestion
    public function withRetry(callable $retryPolicy, int $retryTimes = 1, int $retryInterval = 300): self
```